src/commands/close-project.js

```diff
@@ -0,0 +1,50 @@
+const flow = require('lodash.flow');
+const { URL } = require('url');
+
+const github = require("../lib/github");
+const { withToken } = require('../lib/helpers/yargs/options');
+
+/**
+ * yargs builder function.
+ *
+ * @param {import('yargs').Yargs} yargs - Instance of yargs
+ */
+const builder = yargs => {
+	const baseOptions = flow([withToken]);
+
+	return baseOptions(yargs)
+		.positional('path', {
+            describe: 'URL path to project',
+            type: 'string'
+		});
+};
+
+/**
+ * Close a project.
+ *
+ * @param {object} argv - argv parsed and filtered by yargs
+ * @param {string} argv.token
+ * @param {string} argv.path
+ */
+const handler = async ({ token, path }) => {
+	const url = new URL(path);
+	const pathName = url.pathname;
+	const [ , projectType, name ] = pathName.split('/');
+
+	const { listProjects } = github({
+		personalAccessToken: token
+	});
+
+	const openProjects = await listProjects({ org: name }).catch(error => {
+		throw new Error(`Listing all open projects failed. Response: ${error}.`);
+	});
+
+	console.log(openProjects);
+};
+
+module.exports = {
+	command: "projects:close <path>",
```

Comment:
I'm confused... the handler is listing projects, not closing a project :thinking:

Reply:
Yes, sorry I should have explained. In order to close a project, we need a project ID which as we learned isn't very easy. It's not as simple as getting it from the URL. The way that Sam and I discussed it was that we would:

1. Extract the project type (e.g. repos, orgs, users) and name (e.g. financial-times-sandbox) from the URL passed in through the command line.

2. Use the GitHub API to list all of the `open` projects under that account

3. Find the project to close and project ID by using the URL originally passed in as it will be somewhere in the GitHub API response.

4. Use the GitHub API and project ID to close the project.

Yah, it's not pretty but I couldn't think of anything else. Any suggestions for improvement on this process so welcome!

Reply:
So far I have done points 1 and 2.